packages/tailwindcss-language-server/src/oxide.ts

```diff
@@ -0,0 +1,112 @@
+// This covers the Oxide API from v4.0.0-alpha.1 to v4.0.0-alpha.18
+declare namespace OxideV1 {
+  interface GlobEntry {
+    base: string
+    glob: string
+  }
+
+  interface ScanOptions {
+    base: string
+    globs?: boolean
+  }
+
+  interface ScanResult {
+    files: Array<string>
+    globs: Array<GlobEntry>
+    candidates: Array<string>
+  }
+}
+
+// This covers the Oxide API from v4.0.0-alpha.19+
+declare namespace OxideV2 {
+  interface GlobEntry {
+    base: string
+    pattern: string
+  }
+
+  interface ScanOptions {
+    base: string
+    sources: Array<GlobEntry>
+  }
+
+  interface ScanResult {
+    files: Array<string>
+    globs: Array<GlobEntry>
+    candidates: Array<string>
+  }
+}
+
+interface Oxide {
+  scanDir?(options: OxideV1.ScanOptions): OxideV1.ScanResult
+  scanDir?(options: OxideV2.ScanOptions): OxideV2.ScanResult
+}
+
+async function loadOxideAtPath(id: string): Promise<Oxide | null> {
+  let oxide = await import(id)
+
+  // This is a much older, unsupport version of Oxide before v4.0.0-alpha.1
+  if (!oxide.scanDir) return null
+
+  return oxide
+}
+
+interface GlobEntry {
+  base: string
+  pattern: string
+}
+
+interface ScanOptions {
+  oxidePath: string
+  basePath: string
+  sources: Array<GlobEntry>
+}
+
+interface ScanResult {
+  files: Array<string>
+  globs: Array<GlobEntry>
+  candidates: Array<string>
+}
+
+/**
+ * This is a helper function that leverages the Oxide API to scan a directory
+ * and a set of sources and turn them into files and globs.
+ *
+ * Because the Oxide API has changed over time this function presents a unified
+ * interface that works with all versions of the Oxide API but the results may
+ * be different depending on the version of Oxide that is being used.
+ *
+ * For example, the `sources` option is ignored before v4.0.0-alpha.19.
+ */
+export async function scan(options: ScanOptions): Promise<ScanResult | null> {
+  const oxide = await loadOxideAtPath(options.oxidePath)
+  if (!oxide) return null
+
+  let resultV1: OxideV1.ScanResult | null = null
+  let resultV2: OxideV2.ScanResult | null = null
+
+  try {
+    resultV2 = oxide.scanDir({
+      base: options.basePath,
+      sources: options.sources,
+    })
+  } catch {
+    resultV1 = oxide.scanDir({
+      base: options.basePath,
+      globs: true,
+    })
+  }
```

Comment:
This raises the questions:

- Should we properly version this API? 
- When can we deprecate support for an alpha release version? Should we even support older alpha releases? I worry that if we don't define a supported range right now, this will creep and become an unnecessary burden over time. 

Maybe we can say all alphas are supported for now but once we release something more stable, we'll drop support for it and will expect our users to upgrade? WDYT?

Reply:
Since we have the load path, maybe we could read the `package.json` file to see what version we are working with 🤔

Reply:
1. Should we properly version this API?
Well… we can't do that exactly because it's already released so…

2. When can we deprecate support for an alpha release version?

Good question — I think maybe once we're closer to a more stable API + a month or so we can drop stuff then.

> Since we have the load path, maybe we could read the package.json file to see what version we are working with 

Yeah that is an option 🤔